File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
import re

def solution(s):
    numArr = ['zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine']
    for i in range(len(numArr)):
        num = numArr[i]
        pattern = rf"{num}"
        match = re.findall(pattern, s)
        
        if match:
            replacedS = re.sub(pattern, str(i), s)
            s = replacedS
            
    return int(s)
            

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.js
// function solution(s) {
//     const numbers = ['zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine'];
//     numbers.forEach((v, i) => {
//         let arr = s.split(numbers[i]);
//         s = arr.join(i);
//     })
//     return Number(s);
// }

function solution(s) {
    const numbers = ['zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine'];
    numbers.forEach((v, i) => {
        const regex = new RegExp(`${numbers[i]}`, 'g');
        s = s.replace(regex, i);
    })
    return Number(s);
}